### DISCUSSION TITLE
Regenerate INSTANCE_NAME after migration?

### DISCUSSION THREAD

**Discussion by oclipa** [quartznet/quartznet#1493] [Q&A]
Hi All; we recently migrated our existing quartz tables to new system, where we want to preserve the existing schedules but the instanceids for the new cluster have changed.  This means the entries in the QRTZ_SCHEDULER_STATE table are being (apparently) ignored on the new system because the INSTANCE_NAME values are now wrong.

Is there a way to automatically re-generate/change/fix the INSTANCE_NAME values for existing schedules?

**Comment by lahma** [comment]
You can configure this value for each scheduler, [see the configuration reference](https://www.quartz-scheduler.net/documentation/quartz-3.x/configuration/reference.html#quartz-scheduler-instancename).

**Reply by oclipa** [reply]
Thanks.  Are you suggesting we implement our own InstanceIdGenerator to set the INSTANCE_NAMES for the new system to match those of the old system?

**Reply by lahma** [reply]
You could assign the names/ids used in old environment to new one manually using the configuration. Assuming you have the same node configuration (at least as many nodes as in old environment). If you were using automatic ids it's hard to generate same ids again. Unless you want to manually migrate data in database to some other fixed (predictable id) and assign that then manually.

You are referring id and name interchangeably here.  Instance id is not so important as nodes can come and go, instance name is important so that the nodes participating in cluster are handling correct schedules. This is easy to set via configuration.

**Reply by oclipa** [reply]
Thanks again.  Just to be clear: instance name is the value that appears in the SCHED_NAME column, while instance id is the value that appears in the INSTANCE_NAME column - is that correct?  (of the QRTZ_SCHEDULER_STATE table)

**Reply by lahma** [reply]
Yes, these can be confusing. `SCHED_NAME` is the logical scheduler (WarehouseScheduler) which of there can be multiple nodes (cluster)  `INSTANCE_NAME` (APP-01-WEST-EUROPE).

Scheduler Name, instance name in config, unique within server/process
Instance Id ~= Machine Id, unique within a cluster

These get confusing if you look too much under the hood (database), the configuration reference naming is more understandable, DB has some legacy.

**Reply by oclipa** [reply]
Is there any way to trigger the regeneration of the instance ids?  (i.e. trigger new nodes to be generated)

**Reply by lahma** [reply]
If you start new nodes with AUTO instance id they should get new ids?

**Reply by oclipa** [reply]
OK.  Also, I now note that if an instance id is invalid, it appears that quartz automatically generates a new node.  Thanks for the help.

**Reply by fbrum** [reply]
Hi @lahma .

This was really clarifying and explanatory why the behavior is different between 2.x and 3.x

But part of the confusion we had is that the documentation compared to previous behavior was misleading:
![image](https://user-images.githubusercontent.com/19552674/203512839-c1fa3659-e192-424e-953f-08210ed11520.png)

This is much more direct, simple and clear.

> Scheduler Name, instance name in config, unique within server/process
>Instance Id ~= Machine Id, unique within a cluster
 
Maybe it's just me, but instead of 
>  "...must be unique for all schedulers working.."

we could have
>  "...must be unique within each schedulers working.."

Many thanks
